Userland/Libraries/LibCrypto/Curves/SECPxxxr1.h

```diff
@@ -0,0 +1,700 @@
+        auto u1_buf = TRY(ByteBuffer::create_uninitialized(KEY_BYTE_SIZE));
+        export_big_endian(u1, u1_buf.bytes());
+        auto u2_buf = TRY(ByteBuffer::create_uninitialized(KEY_BYTE_SIZE));
+        export_big_endian(u2, u2_buf.bytes());
+
+        auto p1 = TRY(generate_public_key(u1_buf));
+        auto p2 = TRY(compute_coordinate(u2_buf, pubkey));
+
+        JacobianPoint point1 {
+            import_big_endian(TRY(p1.slice(1, KEY_BYTE_SIZE))),
+            import_big_endian(TRY(p1.slice(1 + KEY_BYTE_SIZE, KEY_BYTE_SIZE))),
+            1u,
+        };
+
+        // Convert the input point into Montgomery form
+        point1.x = to_montgomery(point1.x);
+        point1.y = to_montgomery(point1.y);
+        point1.z = to_montgomery(point1.z);
+
+        VERIFY(is_point_on_curve(point1));
+
+        JacobianPoint point2 {
+            import_big_endian(TRY(p2.slice(1, KEY_BYTE_SIZE))),
+            import_big_endian(TRY(p2.slice(1 + KEY_BYTE_SIZE, KEY_BYTE_SIZE))),
+            1u,
+        };
+
+        // Convert the input point into Montgomery form
+        point2.x = to_montgomery(point2.x);
+        point2.y = to_montgomery(point2.y);
+        point2.z = to_montgomery(point2.z);
+
+        VERIFY(is_point_on_curve(point2));
+
+        JacobianPoint result;
+        point_add(result, point1, point2);
+
+        // Convert from Jacobian coordinates back to Affine coordinates
+        convert_jacobian_to_affine(result);
+
+        // Make sure the resulting point is on the curve
+        VERIFY(is_point_on_curve(result));
+
+        // Convert the result back from Montgomery form
+        result.x = from_montgomery(result.x);
+        result.y = from_montgomery(result.y);
+        // Final modular reduction on the coordinates
+        result.x = modular_reduce(result.x);
+        result.y = modular_reduce(result.y);
+
+        return r.is_equal_to_constant_time(result.x);
+    }
+
+private:
+    constexpr StorageType import_big_endian(ReadonlyBytes bytes)
+    {
+        VERIFY(bytes.size() == KEY_BYTE_SIZE);
+
+        StorageType result = 0u;
+        for (size_t i = 0; i < bytes.size(); i++) {
+            result |= (static_cast<StorageType>(bytes[bytes.size() - i - 1]) << (i * 8));
+        }
+        return result;
+    }
+
+    constexpr void export_big_endian(StorageType value, Bytes bytes)
+    {
+        VERIFY(bytes.size() == KEY_BYTE_SIZE);
+
+        for (size_t i = 0; i < bytes.size(); i++) {
+            bytes[bytes.size() - i - 1] = static_cast<u8>(value & 0xff);
+            value >>= 8;
+        }
+    }
+
+    constexpr StorageType select(StorageType const& left, StorageType const& right, bool condition)
+    {
+        // If condition = 0 return left else right
+        StorageType mask = (StorageType)condition - 1;
+
+        return (left & mask) | (right & ~mask);
+    }
+
+    constexpr StorageTypeX2 multiply(StorageType const& left, StorageType const& right)
+    {
+        return left.wide_multiply(right);
```

Comment:
It is kinda silly we have this wrapper. I originally left it since I was not fond of the idea of changing all the uses.